include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -58,93 +58,103 @@ using pointer_t = vk::SpirvOpaqueType<spv::OpTypePointer,vk::Literal<vk::integra
 
 template<typename Signed>
 [[vk::ext_instruction( spv::OpAtomicSMin )]]
-Signed atomicSMin([[vk::ext_reference]] int32_t ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
+enable_if_t<is_same_v<Signed,uint32_t> || is_same_v<Signed,int32_t>, Signed> atomicSMin([[vk::ext_reference]] int32_t ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
 
 template<typename Signed, typename Ptr_T> // DXC Workaround
 [[vk::ext_instruction(spv::OpAtomicSMin)]]
-typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, Signed>::type atomicSMin(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
+enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<Signed,uint32_t> || is_same_v<Signed,int32_t>), Signed> atomicSMin(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
 
 template<typename Unsigned>
 [[vk::ext_instruction( spv::OpAtomicUMin )]]
-Unsigned atomicUMin([[vk::ext_reference]] Unsigned ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
+enable_if_t<is_same_v<Unsigned,uint32_t> || is_same_v<Unsigned,int32_t>, Unsigned> atomicUMin([[vk::ext_reference]] Unsigned ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
 
 template<typename Unsigned, typename Ptr_T> // DXC Workaround
 [[vk::ext_instruction(spv::OpAtomicUMin)]]
-typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, Unsigned>::type atomicUMin(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
+enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<Unsigned,uint32_t> || is_same_v<Unsigned,int32_t>), Unsigned> atomicUMin(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
 
 template<typename Signed>
 [[vk::ext_instruction( spv::OpAtomicSMax )]]
-Signed atomicSMax([[vk::ext_reference]] Signed ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
+enable_if_t<is_same_v<Signed,uint32_t> || is_same_v<Signed,int32_t>, Signed> atomicSMax([[vk::ext_reference]] Signed ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
 
 template<typename Signed, typename Ptr_T> // DXC Workaround
 [[vk::ext_instruction(spv::OpAtomicSMax)]]
-typename nbl::hlsl::enable_if<is_spirv_type_v<Ptr_T>, Signed>::type atomicSMax(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
+enable_if_t<is_spirv_type_v<Ptr_T> && (is_same_v<Signed,uint32_t> || is_same_v<Signed,int32_t>), Signed> atomicSMax(Ptr_T ptr, uint32_t memoryScope, uint32_t memorySemantics, Signed value);
 
 template<typename Unsigned>
 [[vk::ext_instruction( spv::OpAtomicUMax )]]
-Unsigned atomicUMax([[vk::ext_reference]] uint32_t ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
+enable_if_t<is_same_v<Unsigned,uint32_t> || is_same_v<Unsigned,int32_t>, Unsigned> atomicUMax([[vk::ext_reference]] uint32_t ptr, uint32_t memoryScope, uint32_t memorySemantics, Unsigned value);
```

Comment:
not only does the SPIR-V opcode have `S` in the name, but I also called the parameter `Signed` therefore you should check only for `is_same_v<int32_t>` NOT `uint32_t`

Reverse for functions for intrinsics with `U` in the name / `Unsigned` template parameter

Reply:
Done